robotoff/insights/ocr/core.py

```diff
@@ -109,20 +109,32 @@ def get_source(image_name: str, json_path: str = None, barcode: str = None):
     return "/{}/{}.jpg" "".format("/".join(split_barcode(barcode)), image_name)
 
 
-def ocr_iter(input_str: str) -> Iterable[Tuple[Optional[str], Dict]]:
-    if is_barcode(input_str):
-        image_data = fetch_images_for_ean(input_str)["product"]["images"]
+def ocr_iter_jsonl(stream: TextIO) -> Iterable[Tuple[Optional[str], Dict]]:
+    for line in stream:
+        json_data = json.loads(line)
+
+        if "content" in json_data:
+            source = json_data["source"].replace("//", "/")
+            yield source, json_data["content"]
+
+
+def ocr_iter(source: Union[str, TextIO]) -> Iterable[Tuple[Optional[str], Dict]]:
+    if not isinstance(source, str):
```

Comment:
It seems good to me. However, there are many mypy errors in ocr_iter function, due to the multiple types associated with ocr_iter source parameter. I know some of these errors were already there before your change, but can you however take a look at it? If you don't have time it's alright.